AzFilesHybrid/AzFilesHybrid.psm1

```diff
@@ -4006,11 +4004,118 @@ function Debug-AzStorageAccountEntraKerbAuth {
             $checksExecuted += 1;
             Debug-EntraKerbAdminConsent -StorageAccountName $StorageAccountName -checkResult $checks["CheckAdminConsent"]
         }
+        #
+        #Check Default share and RBAC permissions
+        if (!$filterIsPresent -or $Filter -match "CheckRBAC")
+        {
+            try {
+                $checksExecuted += 1
+                Write-Verbose "CheckRBAC - START"
+
+                $StorageAccountObject = Validate-StorageAccount `
+                    -ResourceGroupName $ResourceGroupName `
+                    -StorageAccountName $StorageAccountName `
+                    -ErrorAction Stop
+                
+                if ($null -eq $StorageAccountObject.AzureFilesIdentityBasedAuth)
+                { 
+                    $checks["CheckRBAC"].Result = "Failed"
+                    $checks["CheckRBAC"].Issue = "AzureFilesIdentityBasedAuth IS NULL"
+                    Write-Error "CheckRBAC - FAILED"
+                }
+                else 
+                {
+                    $DefaultSharePermission = $StorageAccountObject.AzureFilesIdentityBasedAuth.DefaultSharePermission
+                    
+                    if((!$DefaultSharePermission) -or ($DefaultSharePermission -eq 'None'))
+                    {
+                        Debug-RBACCheck -StorageAccountName $StorageAccountName -UserPrincipalName $UserPrincipalName -checkResult $checks["CheckRBAC"]
+                    }

```

Comment:
Let's add an `else` statement logging that access is granted via the default share permission